### PR TITLE
#371,#373: Set up Swagger and document existing endpoints

### PR DESCRIPTION
**Pull request by DanBondarenko**
Issue: 
* https://github.com/keeleinstituut/tv-tolkevarav/issues/371
* https://github.com/keeleinstituut/tv-tolkevarav/issues/373

### CODE REVIEW

**Review by kadmit — COMMENTED** [review]
@DanBondarenko Thank you for the PR. I found a few things that are not matched with the requirements:

1. @OA\Info and @OA\Server annotations are added to App\Http\Controllers\Controller, server url should be left as `/`
2. Dockerfile entrypoint script should contain php artisan l5-swagger:generate to ensure openapi json generation when launching the application

Are there any special reasons for skipping that parts?

File: application/app/Http/Controllers/Controller.php [hunk 1]
```diff
@@ -5,7 +5,26 @@
 use Illuminate\Foundation\Auth\Access\AuthorizesRequests;
 use Illuminate\Foundation\Validation\ValidatesRequests;
 use Illuminate\Routing\Controller as BaseController;
+use OpenApi\Attributes as OA;
 
+#[OA\OpenApi(security: [['WebClientBearerJwt' => []]])]
+#[OA\Server(
+    url: 'https://api.dev.tolkevarav.eki.ee/translation-order/api/v1',
```
[the remark this answers is not in the document]
Comment (DanBondarenko) [reply application/app/Http/Controllers/Controller.php:12]: Thank you for the spotting! I assumed that the "sample annotations" under [issue 371](https://github.com/keeleinstituut/tv-tolkevarav/issues/371) were just an example, not the requirements for the issue.

Therefore when I had to choose the server URI, I used the value which made the most sense to me. The benefit of using the DEV env URI is that, if JWT is configured correctly, testers can use the Swagger UI to manually make requests to the dev environment API. However, if the `/` path is required, I can make that change – @thenouan should give the signal.

I also noticed that I left in a typo with the `/v1` suffix in the URI – fixing asap.

**Comment by DanBondarenko** [comment]
> @DanBondarenko Thank you for the PR. I found a few things that are not matched with the requirements:
> 
> 1. @OA\Info and @OA\Server annotations are added to App\Http\Controllers\Controller, server url should be left as `/`
> 2. Dockerfile entrypoint script should contain php artisan l5-swagger:generate to ensure openapi json generation when launching the application
> 
> Are there any special reasons for skipping that parts?

Well spotted, @kadmit! The point about Dockerfile entrypoint is 100% an oversight of mine. I’ll add it to this PR as well as the others.

File: application/app/Http/Controllers/Controller.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (thenouan) [reply application/app/Http/Controllers/Controller.php:12]: Having server as "/" doesn't put a dependency to infra and leaves free hands to sysadmin to configure the app exposure in infra, for example changing domain to api.dev.something.ee. Now the issue with "/" can be that if service is running under some basepath, e.g "some.domain/some/subpath" and urls are "some.domain/some/subpath/api/v1.." then "/" probably makes the request to the domain and omits the subpath

But we have the APP_URL env variable, which states the basepath for the app, we could use that

[the remark this answers is not in the document]
Comment (DanBondarenko) [reply application/app/Http/Controllers/Controller.php]: Computing the URL from APP_URL env is a great idea and I added it in the latest revision. The only way to access the environment was to define custom annotation, so I ended up doing that (annotation "constants" are only supported for PHPDoc annotations, not attributes).

@kadmit Ready for a re-review.

**Review by kadmit — APPROVED** [review]
@DanBondarenko Thank you for the changes! Good job!

**Review by thenouan — APPROVED** [review]

**Pull request merged**